src/sync.rs

```diff
@@ -218,7 +218,35 @@ impl GILOnceCell<Py<PyType>> {
     /// Get a reference to the contained Python type, initializing it if needed.
     ///
     /// This is a shorthand method for `get_or_init` which imports the type from Python on init.
-    pub(crate) fn get_or_try_init_type_ref<'py>(
+    ///
+    /// # Example: Using `GILOnceCell` to avoid the overhead of importing a class multiple times
+    ///
+    /// ```
+    /// # use pyo3::prelude::*;
+    /// # use pyo3::sync::GILOnceCell;
+    /// # use pyo3::types::{PyDict, PyType};
+    /// # use pyo3::intern;
+    /// #
+    /// #[pyfunction]
+    /// fn create_ordered_dict<'py>(py: Python<'py>, dict: Bound<'py, PyDict>) -> PyResult<Bound<'py, PyAny>> {
+    ///     // Even if this function is called multiple times,
+    ///     // the `OrderedDict` class will be imported only once.
+    ///     static ORDERED_DICT: GILOnceCell<Py<PyType>> = GILOnceCell::new();
+    ///     ORDERED_DICT
+    ///         .get_or_try_init_type_ref(py, "collections", "OrderedDict")?
+    ///         .call1((dict,))
+    /// }
+    ///
+    /// # Python::with_gil(|py| {
+    /// #     let dict = PyDict::new(py);
+    /// #     dict.set_item(intern!(py, "foo"), 42).unwrap();
+    /// #     let fun = wrap_pyfunction!(create_ordered_dict, py).unwrap();
+    /// #     let ordered_dict = fun.call1((&dict,)).unwrap();
+    /// #     assert!(dict.eq(ordered_dict).unwrap());
+    /// # });
+    /// ```
+    ///
+    pub fn get_or_try_init_type_ref<'py>(
```

Comment:
This can use a better name. Maybe just "import"? 🤔

Reply:
I agree, but then maybe it should be implemented for `GILOnceCell<Py<T>> where T: PyTypeInfo` instead of `GILOnceCell<Py<PyType>>`? Otherwise, I would call it `import_type`.

Reply:
I'd be fine with that. I think you want `T: PyTypeCheck` rather than `T: PyTypeInfo`, though.

Reply:
Right, done in https://github.com/PyO3/pyo3/pull/4542/commits/79568f6c267d8973504eb94b19ae0f32a92d066c!